homeassistant/components/ssdp/__init__.py

```diff
@@ -85,6 +86,29 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+_SsdpServiceInfoBase = TypedDict(
+    "_SsdpServiceInfoBase",
+    {
+        "x-homeassistant-matching-domains": set,
+    },
+    total=True,
+)
+
+
+class SsdpServiceInfo(_SsdpServiceInfoBase, total=False):
+    """Prepared info from ssdp entries."""
+
+    # Attributes for accessing info from SSDP response
+    ssdp_location: str
```

Comment:
Should we add extra (non-required) attributes, or should we keep them for a follow-up PR?
```suggestion
    ssdp_location: str
    ssdp_st: str
    ssdp_nt: str
    ssdp_udn: str
    ssdp_usn: str
    ssdp_ext: str
    ssdp_server: str
```